### PR TITLE
Add transmit-disk 4.4.12

### PR DESCRIPTION
**Pull request by phatblat**
Additionally, if **adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not already refused in [closed issues].
- [x] Checked the cask is submitted to [the correct repo].

## Notes 

Transmit Disk is a helper app for Transmit which is separate due to Mac App Store sandboxing restrictions. The helper app is necessary to mount a remote server as a disk in the Finder. I believe it uses [FUSE](https://osxfuse.github.io) to do this as it installs a 2nd bundle to `/Library/Filesystems/transmitdisk.fs`.

The `com.panic.TransmitDisk.pkg` package installs `Transmit Disk.app` into `/tmp/` which doesn't seem like a good permanent home.

I tried setting `customLocation` in choices, which worked to put the .app in `/Applications`, but the installer validation failed. So, instead this uses a `postflight` move the .app into `/Applications`, paired with an `uninstall`/`trash` of the .app.

### CODE REVIEW

File: Casks/transmit-disk.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                 # FIXME: Puts .app in correct place but fails package validation
+                 # {
+                 #   'choiceIdentifier' => 'com.panic.TransmitDisk.pkg',
+                 #   'choiceAttribute'  => 'customLocation',
+                 #   'attributeSetting' => '/Applications',
+                 # },
+               ]
+
+  postflight do
+    system_command 'mv',
```
Comment (vitorgalvao) [inline Casks/transmit-disk.rb]: Don’t use `mv` (also, always use the full path). Instead, use ruby itself (`FileUtils`).

File: Casks/transmit-disk.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                   args: [
+                           '/tmp/Transmit/Transmit Disk.app',
+                           '/Applications/',
```
Comment (vitorgalvao) [inline Casks/transmit-disk.rb]: Use `appdir` so it uses the user’s choice. Also, don’t end directory paths in `/`.

File: Casks/transmit-disk.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  uninstall quit:    [
+                       'com.panic.Transmit',
+                       'com.panic.Transmit.TransmitDisk',
+                       'com.panic.Transmit.TransmitMenu',
+                     ],
+            pkgutil: [
+                       'com.panic.TransmitDisk.pkg',
+                       'com.panic.TransmitDisk.filesystem.tdfuse.pkg',
+                     ],
+            trash:   '/Applications/Transmit Disk.app'
```
Comment (vitorgalvao) [inline Casks/transmit-disk.rb]: `appdir` here as well. Also, use `delete`, `trash` does not work yet.

File: Casks/transmit-disk.rb [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+  zap delete: [
+                '~/Library/Preferences/com.panic.Transmit.TransmitDisk.plist',
+                '/Library/Filesystems/transmitdisk.fs/',
+                '/Applications/Transmit Disk.app',
```
Comment (vitorgalvao) [inline Casks/transmit-disk.rb]: `zap` always runs `uninstall` first. Remove this line.

File: Casks/transmit-disk.rb [hunk 5]
```diff
@@ -0,0 +1,61 @@
+cask 'transmit-disk' do
+  version '4.4.12'
+  sha256 '3a4e3515450e597fc7f0d75f0d53df215bea3b6d9615c0a960f471b1185602ef'
+
+  url "https://download.panic.com/transmit/Transmit%20Disk%20#{version}.pkg.zip"
+  name 'transmit-disk'
+  homepage 'https://panic.com/transmit/'
+
+  depends_on cask: 'transmit'
```
Comment (vitorgalvao) [inline Casks/transmit-disk.rb]: If `transmit` can be installed from the App Store, don’t force it being installed by us.

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

File: Casks/transmit-disk.rb (hunk 1 above)
Comment (phatblat) [inline Casks/transmit-disk.rb]: I don't know if `FileUtils` will work. Is there a way to run it through `sudo`?

```
Error: Permission denied - (/tmp/Transmit/Transmit Disk.app, /Applications/Transmit Disk.app)
```

Installer packages get ran [through `sudo`](https://github.com/Homebrew/brew/blob/d1f599f1ed51976fc76ad19803de04864d12ef2b/Library/Homebrew/cask/lib/hbc/artifact/pkg.rb#L55), so the installed files are owned by root.

```
ls -ld /Applications /tmp /tmp/Transmit /tmp/Transmit/Transmit\ Disk.app                                                                                                                                                                                                          5  18:41:42
drwxrwxr-x+ 163 root        admin        5542 May 27 18:05 /Applications/
lrwxr-xr-x@   1 root        wheel          11 Sep 26  2016 /tmp -> private/tmp
drwxr-xr-x    3 root        wheel         102 May 27 18:40 /tmp/Transmit/
drwxr-xr-x    3 root        wheel         102 May 27 18:40 /tmp/Transmit/Transmit Disk.app/
```

Comment (phatblat) [inline Casks/transmit-disk.rb]: nvm, I found [set_ownership](https://github.com/Homebrew/brew/blob/b6f3399cb3facababf4f6b1e6f7821edc2db554a/Library/Homebrew/cask/lib/hbc/staged.rb#L33)

**Review by vitorgalvao — APPROVED** [review]

**Pull request merged**